src/main/kotlin/com/mparticle/kits/OneTrustKit.kt

```diff
@@ -254,20 +281,25 @@ class OneTrustKit : KitIntegration(), IdentityStateListener {
             Logger.warning(jse, "OneTrust parsing error")
             listOf()
         }.map {
-                    val cookieValue = it.optString("value")
-                    val purpose = it.optString("map")
-                    val regulation = when (purpose) {
-                        CCPAPurposeValue -> ConsentRegulation.CCPA
-                        else -> ConsentRegulation.GDPR
-                    }
-                    if (cookieValue.isNullOrEmpty() && purpose.isNullOrEmpty()) {
-                        Logger.warning("Consent Object is missing value and map: $this")
-                        null
-                    } else {
-                        cookieValue to OneTrustConsent(purpose, regulation)
-                    }
-                }
+            val cookieValue = it.optString("value")
+            val purpose = it.optString("map")
+            val regulation = when (purpose) {
+                CCPAPurposeValue -> ConsentRegulation.CCPA
+                else -> ConsentRegulation.GDPR
+            }
+            if (cookieValue.isNullOrEmpty() && purpose.isNullOrEmpty()) {
+                Logger.warning("Consent Object is missing value and map: $this")
+                null
+            } else {
+                cookieValue to OneTrustConsent(purpose, regulation)
+            }
+        }
             .filterNotNull()
             .associate { it.first to it.second }
     }
+
+    override fun onUserIdentified(user: MParticleUser, previousUser: MParticleUser?) {
+
```

Comment:
I think you want a `processOneTrustConsents()` here. we store consent state at the MPID level, and so we want to make sure whenever MPID changes (and this method gets called when that happens), we are syncing the onetrust state to the user.

Reply:
oh ok perfect ill add it